src/plugins/wmr/plugin.js

```diff
@@ -65,33 +67,42 @@ export default function wmrPlugin({ hot = true } = {}) {
 				before += `const module={${hot ? 'hot:import.meta.hot' : ''}};\n`;
 			}
 
+			const hasEsmKeywords = ESM_KEYWORDS.test(code);
+			const hasExport = code.match(/\bexport\b/);
+
 			// Detect modules that appear to have both JSX and an export, and inject prefresh:
 			// @todo: move to separate plugin.
 			// if (code.match(/\/\*@@prefresh_include\*\//) && code.match(/\bexport\b/)) {
-			if (code.match(/html`[^`]*<([a-zA-Z][a-zA-Z0-9.:-]*|\$\{.+?\})[^>]*>/) && code.match(/\bexport\b/)) {
+			if (code.match(/html`[^`]*<([a-zA-Z][a-zA-Z0-9.:-]*|\$\{.+?\})[^>]*>/) && hasExport) {
 				// if (this.getModuleInfo(id).hasJSX) {
 				hasHot = true;
 				after += '\n' + PREFRESH;
 				// }
 			}
 
-			if (!hasHot) return null;
+			if (!hot || !hasEsmKeywords) return null;
```

Comment:
third time's a charm?

```suggestion
			if (!hasHot && (!hot || !hasEsmKeywords)) return null;
```